### PR TITLE
Create specific error class to signal uplink fetcher errors

### PR DESCRIPTION
**Pull request by nogates**
Hello there,

As described in #1439, we are suggesting here to use a specific class when throwing errors while fetching the schema from Apollo Uplink.

This should allow us to capture those errors using the specific class name, instead of relying on the actual message.

I've decided to transform all the errors in that file (since they are all related to the same process of fetching the schema). Also, please let me know if you think the error class should be placed somewhere else, so we don't need to export it up to the top

Thank you!

Fixes

### CODE REVIEW

File: gateway-js/src/supergraphManagers/UplinkFetcher/index.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import resolvable from '@josephg/resolvable';
 import { SupergraphManager, SupergraphSdlHookOptions } from '../../config';
 import { SubgraphHealthCheckFunction, SupergraphSdlUpdateFunction } from '../..';
 import { loadSupergraphSdlFromUplinks } from './loadSupergraphSdlFromStorage';
+export { UplinkFetcherError } from "./loadSupergraphSdlFromStorage"
```
Comment (trevor-scheer) [inline gateway-js/src/supergraphManagers/UplinkFetcher/index.ts:7]: It's a little funny to just export the error here for the sake of a simpler re-export from `supergraphManagers/index`.

I think _either_:
* This Error should be defined and exported from here (`UplinkFetcher` owns `UplinkFetcherError`, makes sense to me)
* Or, just skip exporting from here and re-export it from the nested path

One could probably argue that the contents of `loadSupergraphSdlFromStorage.ts` (and transitively `outOfBandReporter.ts`) simply belong in the `UplinkFetcher` class which is probably the root cause of ambiguity for "where should the Error class be defined?", though I'm not asking you to resolve that issue in this PR.

Comment (nogates) [reply gateway-js/src/supergraphManagers/UplinkFetcher/index.ts:7]: Yep. I completely agree. I think it's probably cleaner if I define the Error class in the `index`, and just get it from there in `loadSupergraphSdlFromStorage`.
Will change it. Thanks for the feedback

Comment (nogates) [reply gateway-js/src/supergraphManagers/UplinkFetcher/index.ts]: Actually, I was having problems defining `UplinkFetcherError` in the index file, and using it in `loadSupergraphSdlFromStorage`... so I've decided to go with the second option...

However, this makes me wonder, whether the error should belong in a different file (`errors.ts`), and then, export it from there? 

> One could probably argue that the contents of loadSupergraphSdlFromStorage.ts (and transitively outOfBandReporter.ts) simply belong in the UplinkFetcher class which is probably the root cause of ambiguity for "where should the Error class be defined?

**Review by trevor-scheer — APPROVED** [review]
This LGTM. Thanks for the changes! Can you please add an entry to the `gateway-js/CHANGELOG.md`? We'll port this to the `version-0.x` branch when this lands.

cc @cpeacock for awareness since you participated on the linked issue - sounds like there might be an Atlas issue we can close out when this lands?

**Comment by cpeacock** [comment]
pinging @clenfest for awareness

**Pull request merged**